sampler.go

```diff
@@ -199,31 +200,37 @@ func (s *rateLimitingSampler) Equal(other Sampler) bool {
 
 // GuaranteedThroughputProbabilisticSampler is a sampler that leverages both probabilisticSampler and
 // rateLimitingSampler. The rateLimitingSampler is used as a guaranteed lower bound sampler such that
-// every operation is sampled at least once in a time interval defined by the lowerBound. ie a lowerBound
-// of 1.0 / (60 * 10) will sample an operation at least once every 10 minutes.
+// every operation is sampled at least once in a time interval defined by minSamplesPerSecond. ie a
+// minSamplesPerSecond of 1.0 / (60 * 10) will sample an operation at least once every 10 minutes.
 //
 // The probabilisticSampler is given higher priority when tags are emitted, ie. if IsSampled() for both
 // samplers return true, the tags for probabilisticSampler will be used.
 type GuaranteedThroughputProbabilisticSampler struct {
-	probabilisticSampler *ProbabilisticSampler
-	lowerBoundSampler    Sampler
-	tags                 []Tag
-	samplingRate         float64
-	lowerBound           float64
+	probabilisticSampler  *ProbabilisticSampler
+	lowerBoundSampler     Sampler
+	upperBoundRateLimiter utils.RateLimiter
+	tags                  []Tag
+	samplingRate          float64
+	minSamplesPerSecond   float64
+	maxSamplesPerSecond   float64
 }
 
 // NewGuaranteedThroughputProbabilisticSampler returns a delegating sampler that applies both
 // probabilisticSampler and rateLimitingSampler.
 func NewGuaranteedThroughputProbabilisticSampler(
-	lowerBound, samplingRate float64,
+	minSamplesPerSecond, samplingRate, maxSamplesPerSecond float64,
```

Comment:
def a nit, and one that's tedious to fix, but it's annoying that samplingRate is between the SPS params